.github/workflows/r-cmd-check.yml

```diff
@@ -1,29 +1,48 @@
----
-name: R CMD Check, Linters and Spelling
-
 on: push
 
+name: R-CMD-checks
+
 jobs:
-  build:
-    runs-on: [ubuntu-latest]
-    timeout-minutes: 120
+  main:
+    name: ${{ matrix.config.os }} (${{ matrix.config.r }})
+
+    runs-on: ${{ matrix.config.os }}
+
+    timeout-minutes: 30
+
     strategy:
+      fail-fast: false
       matrix:
-        r-version: ['release']
+        config:
+        - {os: macOS-latest,   r: 'release'}
+        - {os: windows-latest, r: 'release'}
+        - {os: ubuntu-latest, r: 'release'}
 
     steps:
-      - uses: actions/checkout@v3
+      - name: Checkout repository
+        uses: actions/checkout@v3
 
-      - name: Set up R ${{ matrix.r-version }}
-        uses: r-lib/actions/setup-r@v2
-        with:
-          r-version: ${{ matrix.r-version }}
-
-      - name: Setup libcurl
+      - name: Setup Pandoc for ${{ matrix.config.os }}
```

Comment:
Naming the steps using a variable makes no sense. This particular step is a Pandoc setup for Windows, not for macOS, thus it should be named "Setup Pandoc for Windows". This way it will be understandable, why this step was skipped when checking on macOS:
![image](https://user-images.githubusercontent.com/7279715/214786434-87bfe1b0-1407-4c4f-be6d-79502b0a63c6.png)

Reply:
True, yeah.

Reply:
It was skipped because the if condition was not met. The names should change and so it will look like the othe os are skipped.

Reply:
<img width="246" alt="image" src="https://user-images.githubusercontent.com/28699471/215393454-e273ff57-50e3-4fec-91d3-1cffe66e80ab.png">